clients/ui/bff/internal/integrations/k8s.go

```diff
@@ -150,95 +151,84 @@ func (kc *KubernetesClient) BearerToken() (string, error) {
 func (kc *KubernetesClient) GetServiceNames() ([]string, error) {
-	//TODO (ederign) we should consider and rethinking listing all services on cluster
-	// what if we have thousand of those?
-	// we should consider label filtering for instance
-
-	serviceList := &corev1.ServiceList{}
-	//TODO (ederign) review the context timeout
-	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Second)
-	defer cancel()
-
-	err := kc.ControllerRuntimeClient.List(ctx, serviceList, &client.ListOptions{})
+	services, err := kc.GetServiceDetails()
 	if err != nil {
-		return nil, fmt.Errorf("failed to list services: %w", err)
-	}
-
-	var serviceNames []string
-	for _, service := range serviceList.Items {
-		if value, ok := service.Spec.Selector["component"]; ok && value == ComponentName {
-			serviceNames = append(serviceNames, service.Name)
-		}
+		return nil, err
 	}
 
-	if len(serviceNames) == 0 {
-		return nil, fmt.Errorf("no services found with component: %s", ComponentName)
+	names := make([]string, 0, len(services))
+	for _, svc := range services {
+		names = append(names, svc.Name)
 	}
 
-	return serviceNames, nil
+	return names, nil
 }
 
 func (kc *KubernetesClient) GetServiceDetails() ([]ServiceDetails, error) {
-	//TODO (ederign) review the context timeout
-	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Second)
-	defer cancel() // Ensure the context is canceled to free up resources
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
 
 	serviceList := &corev1.ServiceList{}
 
-	err := kc.ControllerRuntimeClient.List(ctx, serviceList, &client.ListOptions{})
+	labelSelector := labels.SelectorFromSet(labels.Set{
+		"component": ComponentLabelValue,
+	})
+
+	err := kc.ControllerRuntimeClient.List(ctx, serviceList, &client.ListOptions{
+		LabelSelector: labelSelector,
+	})
 	if err != nil {
 		return nil, fmt.Errorf("failed to list services: %w", err)
 	}
 
 	var services []ServiceDetails
 
 	for _, service := range serviceList.Items {
-		if svcComponent, exists := service.Spec.Selector["component"]; exists && svcComponent == ComponentName {
-			var httpPort int32
-			hasHTTPPort := false
-			for _, port := range service.Spec.Ports {
-				if port.Name == "http-api" {
-					httpPort = port.Port
-					hasHTTPPort = true
-					break
-				}
-			}
-			if !hasHTTPPort {
-				kc.Logger.Error("service missing HTTP port", "serviceName", service.Name)
-				continue
-			}
-
-			if service.Spec.ClusterIP == "" {
-				kc.Logger.Error("service missing valid ClusterIP", "serviceName", service.Name)
-				continue
+		var httpPort int32
+		hasHTTPPort := false
+		for _, port := range service.Spec.Ports {
+			if port.Name == "http-api" {
+				httpPort = port.Port
+				hasHTTPPort = true
+				break
 			}
+		}
+		if !hasHTTPPort {
+			kc.Logger.Error("service missing HTTP port", "serviceName", service.Name)
+			continue
+		}
 
-			displayName := ""
-			description := ""
+		if service.Spec.ClusterIP == "" {
```

Comment:
I know this is code from before, but is there a case where this would ever happen when deploying MR?